Program.cs

```diff
@@ -53,49 +53,93 @@ private static async Task<bool> CheckSingleSha(Manifest m)
             sw.Start();
 
-            if (args.Length != 1)
+            if (args.Length != 2)
             {
-                await Console.Error.WriteLineAsync("Usage: ModlinksShaVerifier [FILE]");
+                await Console.Error.WriteLineAsync("Usage: ModlinksShaVerifier [CURRENT_FILE] [INCOMING_FILE]");
                 return 1;
             }
 
-            var path = args[0];
+            string currentPath = args[0];
 
-            if (!File.Exists(path))
+            if (!File.Exists(currentPath))
             {
-                await Console.Error.WriteLineAsync($"Unable to access {path}! Does it exist?");
+                await Console.Error.WriteLineAsync($"Unable to access current XML file {currentPath}! Does it exist?");
                 return 1;
             }
 
-            var reader = XmlReader.Create(path, new XmlReaderSettings {Async = true});
+            var incomingPath = args[1];
 
+            if (!File.Exists(incomingPath))
+            {
+                await Console.Error.WriteLineAsync(
+                    $"Unable to access incoming XML file {incomingPath}! Does it exist?");
+                return 1;
+            }
+            
+            var currentReader = XmlReader.Create(currentPath, new XmlReaderSettings { Async = true });
+            var incomingReader = XmlReader.Create(incomingPath, new XmlReaderSettings { Async = true });
+            
             var serializer = new XmlSerializer(typeof(Manifest));
 
-            List<Task<bool>> checks = new();
-
-            while (await reader.ReadAsync())
+            Dictionary<string, Links> checkedLinks = new();
+            
+            while (await currentReader.ReadAsync())
             {
-                if (reader.NodeType != XmlNodeType.Element)
+                if (currentReader.NodeType != XmlNodeType.Element)
+                    continue;
+
+                if (currentReader.Name != "Manifest")
                     continue;
 
-                if (reader.Name != "Manifest")
+                var currentManifest = (Manifest?)serializer.Deserialize(currentReader) ?? throw new InvalidDataException();
+                TrimManifest(ref currentManifest);
+                checkedLinks.Add(currentManifest.Name, currentManifest.Links);
+            }
+            
+            List<Task<bool>> checks = new();
+            
+            while (await incomingReader.ReadAsync())
+            {
+                if (incomingReader.NodeType != XmlNodeType.Element)
                     continue;
 
-                var manifest = (Manifest?) serializer.Deserialize(reader) ?? throw new InvalidDataException();
+                if (incomingReader.Name != "Manifest")
+                    continue;
 
-                checks.Add(CheckSingleSha(manifest));
+                var incomingManifest = (Manifest?)serializer.Deserialize(incomingReader) ?? throw new InvalidDataException();
+                TrimManifest(ref incomingManifest);

```

Comment:
Might want to use a custom deserializer for this? Just on Link itself. If you don't feel like it, I can do that part